code/WorkInProgress/rewardsLocker.dm

```diff
@@ -1134,6 +1134,77 @@
 		else
 			boutput(activator, "<span class='alert'>You need to be an AI to use this, you goof!</span>")
 
+
+////////AI CORE FRAMES///////
+
+/datum/achievementReward/ai_bee
+	title = "(AI Core Skin) Bee"
+	desc = "Buzz Buzz!"
+	required_medal = "Bombini is Missing!"
+
+	rewardActivate(mob/activator)
+		if (isAI(activator))
+			var/mob/living/silicon/ai/A = activator
+			if (isAIeye(activator))
+				var/mob/living/intangible/aieye/AE = activator
+				A = AE.mainframe
+			A.coreSkin = "bee"
+			A.update_appearance()
+			return 1
+		else
+			boutput(activator, "<span class='alert'>You need to be an AI to use this, you goof!</span>")
```

Comment:
Absolutely Nope to all this copypaste code. Make an abstract subtype - you might need to change whatever creates rewards to use concretetypesof.